src/store/reducers/storage/requestStorageData.ts

```diff
@@ -0,0 +1,24 @@
+import type {AxiosOptions} from '../../../services/api';
+import type {StorageRequestParams} from '../../../types/api/storage';
+import type {GetState} from '../../defaultStore';
+import {selectCapabilityVersion} from '../capabilities/capabilities';
+
+import {prepareGroupsResponse, prepareStorageResponse} from './utils';
+
+export async function requestStorageData(
+    {version = 'v2', ...params}: StorageRequestParams,
+    options: AxiosOptions,
+    getState?: GetState,
+) {
+    const groupsHandlerVersion =
```

Comment:
wouldnt it be better to create enableSomeFunctionality and hook with
return useGetFeatureVersion('/viewer/sth') > 2;

and to pass these flags to functions that depend on it

instead of mixing redux state operations with api calls by passing getState function